src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,227 @@
+import React from 'react'
+import classnames from 'classnames'
+import { Icon } from '../Icon/Icons'
+import { Link } from '../Link/Link'
+import { Button } from '../Button/Button'
+
+type PaginationProps = {
+  pathname: string // pathname of results page
+  totalPages: number // total items divided by items per page
+  currentPage: number // current page number (starting at 1)
+  maxSlots?: number // number of pagination "slots"
+  onClickNext?: () => void
+  onClickPrevious?: () => void
+  onClickPageNumber?: (
+    event: React.MouseEvent<HTMLButtonElement>,
+    page: number
+  ) => void
+}
+
+const PaginationPage = ({
+  page,
+  isCurrent,
+  pathname,
+  onClickPageNumber,
+}: {
+  pathname: string
+  page: number
+  isCurrent?: boolean
+  onClickPageNumber?: (
+    event: React.MouseEvent<HTMLButtonElement>,
+    page: number
+  ) => void
+}) => {
+  const linkClasses = classnames('usa-pagination__button', {
+    'usa-current': isCurrent,
+  })
+
+  return (
+    <li
+      key={`pagination_page_${page}`}
+      className="usa-pagination__item usa-pagination__page-no">
+      {onClickPageNumber ? (
+        <Button
+          type="button"
+          unstyled
+          className={linkClasses}
+          aria-label={`Page ${page}`}
+          aria-current={isCurrent ? 'page' : undefined}
+          onClick={(event) => {
+            onClickPageNumber(event, page)
+          }}>
+          {page}
+        </Button>
+      ) : (
+        <Link
+          href={`${pathname}?page=${page}`}
+          className={linkClasses}
+          aria-label={`Page ${page}`}
+          aria-current={isCurrent ? 'page' : undefined}>
+          {page}
+        </Link>
+      )}
+    </li>
+  )
+}
+
+const PaginationOverflow = () => (
+  <li
+    className="usa-pagination__item usa-pagination__overflow"
+    role="presentation">
+    <span>…</span>
+  </li>
+)
+
+export const Pagination = ({
+  pathname,
+  totalPages,
+  currentPage,
+  className,
+  maxSlots = 7,
+  onClickPrevious,
+  onClickNext,
+  onClickPageNumber,
```

Comment:
Can you add some unit tests that pass in mock onClick handlers and ensure they are called when clicks happen?